src/notebooks_service.py

```diff
@@ -95,10 +97,29 @@ def health():
 @authenticated
 def whoami(user):
     """Return information about the authenticated user."""
-    return Response(
-        json.dumps(user, indent=1, sort_keys=True),
-        mimetype='application/json',
+    headers = {auth.auth_header_name: 'token {0}'.format(auth.api_token)}
+    info = json.loads(
+        requests.request(
+            'GET',
+            '{prefix}/users/{user[name]}'.format(
+                prefix=auth.api_url, user=user
+            ),
+            headers=headers
+        ).text
     )
+    return jsonify(info)
+
+
+@authenticated
+@app.route(urljoin(SERVICE_PREFIX, 'ui/'), defaults={'path': ''})
+@app.route(urljoin(SERVICE_PREFIX, 'ui/<path:path>'), methods=['GET'])
+def ui(path):
+    """Route for serving a UI for managing running servers."""
+    if not path:
+        path = "index.html"
+    return send_from_directory('../static', filename=path)
```

Comment:
can we let flask to serve the static files directly? 

`app=Flask(__name__, static_url_path='ui', static_folder='static')`

Reply:
For production purposes we should serve it via nginx anyway.

Reply:
I did not know how to apply the authenticated decorator to serving static files when Flask is configured this way. If you can tell me how to do that, we can set up the server this way.